### DISCUSSION TITLE
introduction of error bit via Modbus

### DISCUSSION THREAD

**Discussion by GLT-IN** [emsesp/EMS-ESP32#2300] [Q&A]
### DESCRIPTION

Hi,
some devices from other manufacturer are not able to interpret datatype "string" like last error.
Is it possible to introduce a single bit to report there is an active error via modbus?

Best regards,
Markus

### REQUESTED INFORMATION

_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Searched the issue in [issues](https://github.com/emsesp/EMS-ESP32/issues)
- [x] Searched the issue in [discussions](https://github.com/emsesp/EMS-ESP32/discussions)
- [x] Searched the issue in the [docs](https://docs.emsesp.org/Troubleshooting/)
- [x] Searched the issue in the [chat](https://discord.gg/3J3GgnzpyT)
- [x] Provide the System information in the area below, taken from `http://<IP>/api/system`

**Comment by MichaelDvP** [comment]
lasterror shows the newest entry in boiler error log, not an actual error.
`boiler/servicecodenumber` should give you the actual erro code, but also normal status codes, eg. 200-heating mode, 201 dhw-mode, 208-chimny-sweeper-mode, 209-service-mode, etc.
For the functional message codes from thermostat (communication issues, Axx, etc) we have no entity, only the thermostat last erro as string.
On any error in the bus system there is a message type 0xBF sent with reporting device, id, ascii code and codenumber. But this message is only sent on error and on error-clear, trying to read this message results in `<empty>`, so we can't provide it as constant entity.

**Comment by GLT-IN** [comment]
Hi,
can you share the full list of servicecodenumber please?
I dont find them in user manuals.

Best regards,
Markus
